src/logworker.cpp

```diff
@@ -112,14 +114,23 @@ namespace g3 {
 
    void LogWorker::addWrappedSink(std::shared_ptr<g3::internal::SinkWrapper> sink) {
       auto bg_addsink_call = [this, sink] {_impl._sinks.push_back(sink);};
-      auto token_done = g3::spawn_task(bg_addsink_call, _impl._bg.get());
-      token_done.wait();
+      g3::spawn_task(bg_addsink_call, _impl._bg.get());
    }
 
    std::unique_ptr<LogWorker> LogWorker::createLogWorker() {
       return std::unique_ptr<LogWorker>(new LogWorker);
    }
 
+   void LogWorker::removeWrappedSink(std::shared_ptr<g3::internal::SinkWrapper> sink) {
+      auto bg_removesink_call = [this, sink] {
+         _impl._sinks.erase(std::remove(_impl._sinks.begin(), _impl._sinks.end(), sink), _impl._sinks.end());
+
+         // TODO: to be added to the sink API
+         //sink->flush();
+      };
+      g3::spawn_task(bg_removesink_call, _impl._bg.get());
```

Comment:
See above regarding using `send(...)` instead. 

That said. I do consider these two functions (add/sink) to be inherently different. Where the `add` gives a sink that is currently empty and the state is known. 

`remove(...)` removes a sink but the sink will continue to process the messages until it's done. 

There's a fine balance between `flexibility` and `robustness`.  I think the `remove` should provide the only guarantees it can provide. 
1.It is removed from the sink list (done here)
2.No more messages are to be processed once it's removed (not yet done). 

I.e. I'm fine with **no** guarantees regarding destruction of the sink. Since the user can grab the sink that guarantee is impossible to give.  However I really think the 2.guarantee needs to be given. 

I.e. last lines should read (pseudo code0
```
send(bg_removesink_call);
sink->flush())
```